frontend/src/main/components/RequestType/RequestTypeTable.js

```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import OurTable, { ButtonColumn } from "main/components/OurTable";
+
+import { useBackendMutation } from "main/utils/useBackend";
+import {
+  cellToAxiosParamsDelete,
+  onDeleteSuccess,
+} from "main/utils/RequestTypeUtils";
+import { useNavigate } from "react-router-dom";
+import { hasRole } from "main/utils/currentUser";
+
+export default function RequestTypeTable({ requesttype, currentUser }) {
+  const navigate = useNavigate();
+
+  const editCallback = (cell) => {
+    navigate(`/requesttypes/create/${cell.row.values.id}`);
+  };
+
+  // Stryker disable all : hard to test for query caching
+
+  // when delete success, invalidate the correct query key (depending on user role)
+  const apiEndpoint = hasRole(currentUser, "ROLE_PROFESSOR")
+    ? "/api/requesttypes/all"
+    : "/api/requesttypes/all";
+
+  const deleteMutation = useBackendMutation(
+    cellToAxiosParamsDelete,
+    { onSuccess: onDeleteSuccess },
+    [apiEndpoint],
+  );
+  // Stryker restore all
+
+  // Stryker disable next-line all : TODO try to make a good test for this
+  const deleteCallback = async (cell) => {
+    deleteMutation.mutate(cell);
+  };
+
+  const columns = [
+    {
+      Header: "id",
+      accessor: "id", // accessor is the "key" in the data
+    },
+    {
+      Header: "Request Type",
+      accessor: "requestType",
+    },
+  ];
+
+  //since all admins have the role of a user, we can just check if the current user has the role ROLE_USER
+  if (hasRole(currentUser, "ROLE_USER")) {
+    columns.push(
+      ButtonColumn("Delete", "danger", deleteCallback, "RequestTypeTable"),
+    );
+  }
```

Comment:
This comment does not make sense.  It would be like saying: All people that have a ticket to the concert are humans, so we don't need to check for tickets.  Just let them in if they are humans.

If that happened at a concert, do you see the problem?

If you leave the code like this, then ALL users get the delete button.  If that's what you intend, then so be it, but leave the admins out of it.

If only admins are supposed to get the delete button then you need to check for ROLE_ADMIN, not ROLE_USER.

What does the issue say?  Who is supposed to be able to delete on this page?